platform/web/request.go

```diff
@@ -61,88 +83,43 @@ func DecodeWithJSONSchema(request *http.Request, model interface{}, filePath str
+func DecodeWithEmbeddedJSONSchema(request *http.Request, model interface{}, jsonSchema []byte, options DecoderOptions) error {
 	body, err := io.ReadAll(request.Body)
 	if err != nil {
 		if err.Error() == ErrRequestBodyTooLargeMessage {
-			return fmt.Errorf("%w", NewErrRequestBodyTooLarge())
+			return fmt.Errorf("%w", ErrRequestBodyTooLarge)
 		}
-		return err
+		return fmt.Errorf("could not read request body: %v", err)
 	}
 	request.Body = io.NopCloser(bytes.NewBuffer(body))
-	payload := gojsonschema.NewBytesLoader(body)
-
-	result, err := gojsonschema.Validate(jsonSchema, payload)
-	if err != nil {
-		return fmt.Errorf("%w: %v", NewErrBadRequestResponse(nil), err)
-	}
 
-	if !result.Valid() {
-		additionalPropertyErrors := make(ErrorDetails)
-		requiredPropertyErrors := make(ErrorDetails)
-		requestContentErrorDetails := make(ErrorDetails)
-
-		for _, desc := range result.Errors() {
-			propertyName := desc.Field()
-			if propertyName == "(root)" {
-				details, ok := desc.Details()["property"]
-				if ok {
-					propertyName = fmt.Sprintf("%v", details)
-				}
-			}
-
-			switch desc.Type() {
-			case "additional_property_not_allowed":
-				additionalPropertyErrors[propertyName] = desc.Description()
-			case "required":
-				requiredPropertyErrors[propertyName] = desc.Description()
-			default:
-				requestContentErrorDetails[propertyName] = desc.Description()
-			}
+	if err := jsonschema.ValidateJSON(body, jsonSchema); err != nil {
+		switch unwrappedErr := errors.Unwrap(err).(type) {
+		case jsonschema.AdditionalPropertyError:
+			return fmt.Errorf("jsonschema validation error: %w", NewErrBadRequestResponse(unwrappedErr.Details))
+		case jsonschema.MissingPropertyError:
+			return fmt.Errorf("jsonschema validation error: %w", NewErrBadRequestResponse(unwrappedErr.Details))
+		case jsonschema.ValidationError:
+			return fmt.Errorf("jsonschema validation error: %w", NewErrInvalidRequestBodyContent(unwrappedErr.Details))
 		}
-
-		if len(additionalPropertyErrors) > 0 {
-			errDetails := additionalPropertyErrors
-			for property, errMessage := range requestContentErrorDetails {
-				errDetails[property] = errMessage
-			}
-
-			return fmt.Errorf("the request body contains unknown keys: %w", NewErrBadRequestResponse(errDetails))
-		}
-
-		if len(requiredPropertyErrors) > 0 {
-			errDetails := requiredPropertyErrors
-			for property, errMessage := range requestContentErrorDetails {
-				errDetails[property] = errMessage
-			}
-
-			return fmt.Errorf("the request body contains unknown keys: %w", NewErrBadRequestResponse(errDetails))
-		}
-
-		return fmt.Errorf("%w", NewErrInvalidRequestBodyContent(requestContentErrorDetails))
 	}
 
-	if err := decode(request, model, options); err != nil {
-		return err
+	err = json.Unmarshal(body, &model)
+	if err != nil {
+		return fmt.Errorf("json is valid against json schema, but cannot be unmarshalled to the provided struct: %v", err)
```

Comment:
So it does not change behaviour it seems, but I'm not sure if we should return an error we can match on here, as the only reason this can fail is that the struct we provide to the function is not matching the JSON schema, and thus should be a 500.